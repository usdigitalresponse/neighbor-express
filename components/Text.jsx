const Text = ({ block }) => (<section className="grid-container usa-section">
  <div className="grid-row grid-gap">
    {
      block.image && !block.data?.includes('image:right') &&
      <div className="tablet:grid-col-4">
        <img src={block.image} />
      </div>
    }
    <div className="tablet:grid-col-8 usa-prose">
      <h2 className="font-heading-xl margin-top-0 tablet:margin-bottom-0">
        {block.title}
      </h2>
      <div>
        {block.body_markdown}
      </div>
    </div>
    {
      block.image && block.data?.includes('image:right') &&
      <div className="tablet:grid-col-4">
        <img src={block.image} />
      </div>
    }
  </div>
</section>
)



export default Text;