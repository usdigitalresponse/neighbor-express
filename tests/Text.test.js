import { shallow } from "enzyme";
import Text from "../components/Text.jsx";

describe("Text block", () => {
	const block = {
		title: 'aaa',
		body_markdown: 'bbb',
	}
	it('shows header', () => {
    const text = shallow(<Text block={block} />);

    expect(text.find("h2").text()).toEqual("aaa");
  });

  it('shows body', () => {
    const text = shallow(<Text block={block} />);
    expect(text.contains(<div>bbb</div>)).toEqual(true);
  });

  it('shows image', () => {
  	const block_with_image = {
  		...block,
  		image: 'exampleurl.com',
  		alt: 'test alt text'
  	}
    const text = shallow(<Text block={block_with_image} />);
    expect(text.contains(<img src='exampleurl.com' alt='test alt text'/>)).toEqual(true);
  });
});