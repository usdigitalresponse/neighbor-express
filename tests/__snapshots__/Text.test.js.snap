// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text block renders correctly 1`] = `
<Content
  block={
    Object {
      "body_markdown": "bbb",
      "title": "aaa",
    }
  }
>
  <div
    className="grid-row grid-gap"
  >
    <div
      className="tablet:grid-col-12 usa-prose"
    >
      <h2
        className="font-heading-xl margin-top-0 tablet:margin-bottom-0"
      >
        aaa
      </h2>
      <div>
        bbb
      </div>
    </div>
  </div>
</Content>
`;
